kyo-prelude/shared/src/main/scala/kyo/Batch.scala

```diff
@@ -36,8 +31,8 @@ object Batch:
       * This method allows for efficient batching of operations by processing multiple inputs at once, while still providing individual
       * results for each input.
       */
-    inline def source[A, B, S](f: Seq[A] => (A => B < S) < S)(using inline frame: Frame): A => B < Batch[S] =
-        (v: A) => ArrowEffect.suspend[B](erasedTag[S], Call(v, f))
+    inline def source[A, B, S](f: Seq[A] => (A => B < S) < S)(using inline frame: Frame): A => B < (Batch & S) =
+        (v: A) => ArrowEffect.suspendAndMap(Tag[Batch], Call(v, f))(identity)
```

Comment:
The trick is here: the continuation will receive a computation of type `B < (Batch & S)` containing `S` effects from the source. We just flatten need to flatten (`identity`).